### ISSUE TITLE
Draw, select and delete on multible individual cells at once

### ISSUE THREAD

**Issue by golddotasksquestions** [aseprite/aseprite#2002]
This feature has been requested many times over the past years ( https://github.com/aseprite/aseprite/issues/1215, https://github.com/aseprite/aseprite/issues/56) but all issues have been closed with a reference to [#99](https://github.com/aseprite/aseprite/issues/99)

But #99 does not solve this problem at all. 
If I link already existing cells, all my individual cells are merged into the last selected one. This is not the same as drawing on all cells! Linking literally destroys already existing work if the cells were previously unlinked. Besides, Linking would be an additional redundant step, once I already selected the cells I would like to work on.

A much needed behavior would be to be able to draw, select, move and delete on all currently selected cells. (Ctrl-left mouse button in addition to drag select is needed for the same and other reasons see other issue [here](https://github.com/aseprite/aseprite/issues/182, https://github.com/aseprite/aseprite/issues/97))

Proposed workflow would be like this:

- **1** Select the cells you want to operate on (ideally via leftclick, leftdrag, or Ctrl leftclick if cells are not adjacent)
 
- **2a** then draw/nudge/translate/delete a already selected area in the cells on all selected cells simultaneously. ([similar to this](https://www.aseprite.org/docs/move-tool/move-multiple-cels.gif))
 
- **2b** or use the lasso tool to select a subsection within in the cell you want to manipulate, and then nudge/translate/draw/delete on all selected cells simultaneously

Doing the exact same operation on many many cells over and over again is really pointless and tiring and just makes me hate this otherwise really awesome tool.

### Aseprite and System version

* Aseprite version: 1.2.9 installer
* System: Windows 64

**Comment by snarlynarwhal** [comment]
I too think this is quite necessary. I do not want to spend 10 minutes editing 10 layers by 4 frames, aka 40 edits, when it should take 10 seconds.

**Comment by golddotasksquestions** [comment]
Meanwhile, does anyone have a workaround?
The thing I constantly need to do is 
**I have to delete or flip a small area of a cell, for all cells or most cells of my animation.**

Right now this involves clicking on each cell, selecting lasso tool, drawing a selection of the area I want to flip or delete, press delete.
Unfortunately, not only can I not do this operation once at how many cells I please, the selection I made with my lasso or rectangular selection tool disappears right after I press delete. So I have to manually make that selection **again** and **again** for each and every cell. 
If you do this once a year on 4 cells it's fine. If you want to work professionally on a few hundred or thousand cells that's simply unfeasible. Even on 20 cells it is quite infuriating.

**Comment by christiansgao** [comment]
I totally agree, if we can make a feature where we can select multiple layers or even designate area on the screen to import a different animation that would make life much easier.

**Comment by dacap** [comment]
Hi everybody! This is a known issue and pretty critical stuff. I think the ability to apply a "transformation in multiple cels" is the first thing to do.

**Comment by dacap** [comment]
Just in case, this is related to #1459

**Comment by dacap** [comment]
The ability to scale/rotate multiple cels will be available in the next release, Aseprite v1.2.14

**Comment by christiansgao** [comment]
nice!

**Comment by oceanhahn** [comment]
Related to #1729.

**Comment by VelocityRa** [comment]
> This is a known issue and pretty critical stuff

Any news on this? It's been 3 years.

**Comment by celojevic** [comment]
yeah u still cant draw on multiple layers at the same time, unless im missing something?

**Comment by VelocityRa** [comment]
Is it challenging to implement?
I'm curious about the code aspect. The project seems actively worked and in my view this is a really important feature for animation, so I assume it's a complicated one to introduce to the codebase seeing as it's still not done after 4 years.

**Comment by intervolver** [comment]
Please

**Comment by domgetter** [comment]
Perhaps some more concrete suggestions can help move this along.

We should break down this feature per tool:

For the pencil tool:

Make a checkbox next to the Pixel-perfect checkbox labelled "Multi-cel".  When checked, if multiple cels are selected across 1 or more frames, place the pixels on those cels which are visible and unlocked.

Then gate this capability behind an experimental flag so we can see if it works well and we can extend the ideas to the other tools like line, rectangle, etc.

**Comment by golddotasksquestions** [comment]
My biggest need is to have this with the selection tool, so I can move and delete the area I have selected on multiple layers at once.

For moving or deleting a part of a character. For example the head/arm/legs/weapons with many overlays (like clothing, or different armor or weapon loadouts, etc)

**Comment by Gasparoken** [comment]
Just in case, as a reminder: currently is possible to _select an area > select specific range of cels on the time line > transform the pixels of the selection (or delete the content) > clear the selection_ (via 'Esc' key or click anywhere in the editor) after the 'clear selection' action the changes will apply on all cels.
Current inconvenient: Inactive layers are only updated until after the artist deselects them.
Draw on multiple cels isn't allowed yet.

**Comment by golddotasksquestions** [comment]
@Gasparoken
Thank you for the tip, you are correct. However the current implementation is terribly counter intuitive and the opposite of WYSIWYG and user friendliness.

**The** a core feature of the selecting and transforming editing ability is **to see** what you are doing. See where you are moving this selected piece and how you are transforming it. Without this transformation being "real time", there is little point to it at all.

Add to that how I can't first select the layers and frames I want, and then the pixels. This means I never can do this multi-layer, multi-frame transformation on my current selection. I _always_ have to reselect what I had already selected after I drew the selection lasso loop. Terribly annoying!

So you could say the feature is already there, but it is so terrible in practical use, it might as well not be.

**Comment by dacap** [comment]
Hi there, some redesign is being thought for the timeline UI (e.g. 4024), still we need to think how to enable multiple cel editing.

The “check box” idea was something planned (like a “timeline selection lock”, only visible when we select multiple cels). Not exactly a “check box” but a similar idea. I don’t have the mockup at hand but probably something to redesign from scratch anyway.

**Comment by dacap** [comment]
> Add to that how I can't first select the layers and frames I want, and then the pixels

Just as a workaround in the current version of Aseprite, you can check the *Edit > Preferences > Timeline > Keep selection* option:

<img width="932" alt="image" src="https://github.com/aseprite/aseprite/assets/39654/94088b52-be11-4b24-b254-8d09680f4124">

In this way the Timeline selection is kept until you press the <kbd>Escape</kbd> key.
